machine.go

```diff
@@ -1106,9 +1109,9 @@ func (h *Headscale) EnableAutoApprovedRoutes(machine *Machine) error {
 		}
 	}
 
-	for _, approvedRoute := range approvedRoutes {
-		approvedRoute.Enabled = true
-		err = h.db.Save(&approvedRoute).Error
+	for i, approvedRoute := range approvedRoutes {
+		approvedRoutes[i].Enabled = true
+		err = h.db.Save(&approvedRoutes[i]).Error
```

Comment:
Why?

Reply:
The linters were complaining (rightfully so) of a common pitfall of Go. The pointer of `approvedRoute` is always be the same, so just in case we use here the index.